monkey/monkey_island/cc/repository/agent_binary_repository.py

```diff
@@ -0,0 +1,29 @@
+import logging
+from typing import BinaryIO
+
+from . import AgentRetrivalError, FileRetrivalError, IAgentBinaryRepository, IFileRepository
+
+logger = logging.getLogger(__name__)
+
+AGENTS = {"linux": "monkey-linux-64", "windows": "monkey-windows-64.exe"}
+
+
+class AgentBinaryRepository(IAgentBinaryRepository):
+    def __init__(self, file_repository: IFileRepository):
+        self._file_repository = file_repository
+
+    def get_linux_binary(self) -> BinaryIO:
+        try:
+            agent_binary = self._file_repository.open_file(AGENTS["linux"])
+            return agent_binary
+        except FileRetrivalError as err:
+            logger.error(f"An error has occured retriving linux agent binary: {err}")
+            raise AgentRetrivalError(f"It couldn't retrieve linux agent binary: {err}")
+
+    def get_windows_binary(self) -> BinaryIO:
```

Comment:
This whole method is almost the exact same as the `get_linux_binary`

Reply:
https://github.com/guardicore/monkey/pull/1986#discussion_r890238369

Reply:
I don't see how is that comment related. This is a concrete implementation, you can create a private method:
```
def __get_binary(self, filename) -> BinaryIO:
        try:
            agent_binary = self._file_repository.open_file(filename)
            return agent_binary
        except FileRetrivalError as err:
            raise AgentRetrivalError(
                f"An error occurred while retrieving the {filename} agent binary: {err}"
            )
```
Boom, there's no more duplication